src/unit/suggestions.rs

```diff
@@ -127,7 +127,7 @@ impl Instruction {
         })
     }
 
-    pub fn describe_trap_error(&self, registers: &Registers) -> Option<TrapErrorDescription> {
+    pub fn describe_trap_error(&self, registers: &dyn Registers) -> Option<TrapErrorDescription> {
```

Comment:
These methods don't really need to be performant, but I usually write something like this instead when possible:

```suggestion
    pub fn describe_trap_error<R: Registers>(&self, registers: &R) -> Option<TrapErrorDescription> {
```